django_irods/views.py

```diff
@@ -131,22 +127,21 @@ def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=T
             is_zip_request = True
             daily_date = datetime.datetime.today().strftime('%Y-%m-%d')
             output_path = "zips/{}/{}/{}.zip".format(daily_date, uuid4().hex, path)
-            if res.is_federated:
-                irods_path = os.path.join(res.resource_federation_path, path)
-                irods_output_path = os.path.join(res.resource_federation_path, output_path)
-            else:
-                irods_path = path
-                irods_output_path = output_path
+
+            # when res.resource_federation_path is empty, the following statements would be
+            # equivalent to irods_path = path and irods_output_path = output_path
+            irods_path = os.path.join(res.resource_federation_path, path)
+            irods_output_path = os.path.join(res.resource_federation_path, output_path)
 
         store_path = u'/'.join(split_path_strs[1:])  # data/contents/{path-to-something}
         if res.is_folder(store_path):  # automatically zip folders
             is_zip_request = True
             daily_date = datetime.datetime.today().strftime('%Y-%m-%d')
             output_path = "zips/{}/{}/{}.zip".format(daily_date, uuid4().hex, path)
-            if res.is_federated:
-                irods_output_path = os.path.join(res.resource_federation_path, output_path)
-            else:
-                irods_output_path = output_path
+            # when res.resource_federation_path is empty, the following statements would be
+            # equivalent to irods_output_path = output_path
+            irods_output_path = os.path.join(res.resource_federation_path, output_path)
```

Comment:
This should be `irods_output_path = res.irods_path`.  I may have pointed you to the wrong comment earlier.  

This is the one I intended to share.
https://github.com/hydroshare/hydroshare/pull/3559/files/bc43f7e57631465dd9ff8d073152d8060f224be5#r319084487

@pkdash and I talked offline about using `res.irods_path` and he said it'll resolve to the correct path whether it's a resource on a federation or not.

Reply:
@sblack-usu `res.irods_path` returns the root path for the resource that includes resource uuid, but in this case ```output_path``` already contains the resource uuid, so I cannot join it with ```res.irods_path``` but should directly join with ```resource_federation_path```. Hope this makes sense, but let me know if it does not.

Reply:
ok, makes sense after re reading the comment, now.